### PR TITLE
Router route APIs

### PR DESCRIPTION
**Pull request by tshihad**
## Code changes
- Appending host and base URL is moved into `do` function since this is a redundant operation
- Added `id` to `SuccessOrErrorMessage` struct
- Fix UTs
- Added Router route APIs
- UT (TODO)

### CODE REVIEW

**Review by manjunath-batakurki — DISMISSED** [review]
LGTM.

File: pkg/client/plans.go [hunk 1]
```diff
@@ -16,22 +15,13 @@ type PlansAPIService struct {
 	Cfg    Configuration
 }
 
-/*
```
Comment (gandharvas) [inline pkg/client/plans.go:19]: Why is this comment being removed? any specific reason?

File: pkg/client/api.go [hunk 2]
```diff
@@ -49,7 +49,7 @@ func (a *api) do(ctx context.Context, request interface{}, queryParams map[strin
 	if a.path == "" || a.method == "" || a.client == nil || a.jsonParser == nil {
 		panic("api not properly configured")
 	}
-
+	a.path = a.client.getHost() + "/" + consts.VmaasCmpAPIBasePath + "/" + a.path
```
Comment (gandharvas) [inline pkg/client/api.go:52]: would recommend using fmt.Sprintf() here just to maintain consistency

File: pkg/client/instances.go [hunk 3]
```diff
@@ -386,103 +380,13 @@ func (a *InstancesAPIService) SuspendAnInstance(ctx context.Context,
...
-/*
-InstancesAPIService
-Undo the delete of an instance that is in pending removal state
- * @param ctx context.Context - for authentication, logging, cancellation, deadlines, tracing, etc.
- 	Passed from http.Request or context.Background().
- * @param serviceInstanceID
- * @param instanceID
-@return models.GetInstanceResponse
-*/
-func (a *InstancesAPIService) UndoDeleteOfAnInstance(ctx context.Context,
```
Comment (gandharvas) [inline pkg/client/instances.go:398]: is this method not being used now?

Comment (tshihad) [reply pkg/client/instances.go:398]: No

File: pkg/client/plans.go (hunk 1 above)
Comment (tshihad) [reply pkg/client/plans.go:19]: Removed by mistake, will adds back

File: pkg/client/api.go (hunk 2 above)
Comment (tshihad) [reply pkg/client/api.go:52]: '+' have better perfomance then Sprintf, but changing to maintain consistency

**Review by manjunath-batakurki — DISMISSED** [review]
LGTM

**Review by manjunath-batakurki — APPROVED** [review]
LGTM

**Review by gandharvas — APPROVED** [review]

**Pull request merged**